### PR TITLE
gh-98040: Remove find_loader, find_module and other deprecated APIs

### PR DESCRIPTION
**Pull request by warsaw**
* Issue: gh-98040

### CODE REVIEW

**Comment by warsaw** [comment]
Still lots of test failures and probably more things to remove, but it's a start!

**Comment by warsaw** [comment]
@brettcannon @ericsnowcurrently Well, this branch is already getting pretty big.  Pull a thread, unravel a sweater.  There's still some things to do (see the ticket), but it's not too late to provide feedback.  I was pretty aggressive in removing things that seemed to be obsolete or testing obsolete APIs, so hopefully I wasn't too eager to remove things.  Feedback welcome!

**Comment by brettcannon** [comment]
Should this PR be a draft until the tests are passing?

**Comment by warsaw** [comment]
> Should this PR be a draft until the tests are passing?

Yes, done.  The blocker is being worked on upstream so I think pip 23.0 will likely unblock this PR.  In the meantime, I'll keep the branch clean as is.

**Comment by ericsnowcurrently** [comment]
> Well, this branch is already getting pretty big. Pull a thread, unravel a sweater. 

How much work would it be to move the `imp` removal to a separate PR?  If a lot, don't worry about it.  Either way, I am planning on reviewing these changes.

**Comment by warsaw** [comment]
> How much work would it be to move the imp removal to a separate PR? If a lot, don't worry about it. Either way, I am planning on reviewing these changes.

It was kind of tricky, but let me take another look.  I agree it would be better to split these into smaller PRs.

**Comment by ericsnowcurrently** [comment]
Thanks for taking another look!  I know what a pain it can be to split up a big PR.

**Comment by warsaw** [comment]
This branch is close enough now that I'll move it out of draft.  @ericsnowcurrently and @brettcannon It's probably ready for review soon.

File: Doc/reference/import.rst [hunk 1]
```diff
@@ -333,6 +333,9 @@ modules, and one that knows how to import modules from an :term:`import path`
    Use of :meth:`~importlib.abc.MetaPathFinder.find_module` by the import system
    now raises :exc:`ImportWarning`.
 
+.. versionchanged:: 3.12
+   ``find_module()`` has been removed.  Use ``find_spec()`` instead.
```
Comment (arhadthedev) [inline Doc/reference/import.rst]: Should `:meth:` be used here? The Netlify preview is unavailable so I cannot check it myself.

```suggestion
   ``find_module()`` has been removed.  Use :meth:`find_spec` instead.
```

Comment (warsaw) [reply Doc/reference/import.rst]: Yep, thanks!

File: Doc/library/importlib.rst [hunk 2]
```diff
@@ -258,28 +235,6 @@ ABC hierarchy::
                                      +-- SourceLoader
 
 
-.. class:: Finder
```
Comment (AA-Turner) [inline Doc/library/importlib.rst:261]: On new lines 245 & 281, could you change:

```diff
- No longer a subclass of :class:`Finder`.
+ No longer a subclass of :class:`!Finder`.
```

this is needed as the cross-reference target has been removed here. GitHub won't let me do an in-line suggestion, sorry!

**Review by AA-Turner — COMMENTED** [review]
There's quite a few places in the docs where we need to update cross references as of this PR (mainly pre-fixing with an `!`) -- though as they're on lines too far from the changes I can't do inline suggestions. 

Would you be happy for me to push to this branch with said docs cross-referencing changes?

A

**Comment by warsaw** [comment]
> Would you be happy for me to push to this branch with said docs cross-referencing changes?

Yes, this would be very helpful, thanks.

**Comment by warsaw** [comment]
> Done, thank you!

Thank you!

**Review by Yhg1s — APPROVED** [review]

**Pull request merged**